ravenpy/extractors/forecasts.py

```diff
@@ -19,6 +21,12 @@
 
 LOGGER = logging.getLogger("PYWPS")
 
+# Do not remove the trailing / otherwise `urljoin` will remove the geoserver path.
+# Can be set at runtime with `$ env RAVENPY_THREDDS_URL=https://xx.yy.zz/geoserver/ ...`.
+THREDDS_URL = os.environ.get(
+    "RAVENPY_THREDDS_URL", "https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/"
+)
```

Comment:
The resolved `THREDDS_URL` should be checked and add the missing trailing `/` if needed to avoid hard to find errors if `RAVENPY_THREDDS_URL=https://xx.yy.zz/geoserver` (no trailing `/`) was used by mistake. The same comment applies anywhere `urljoin` is used with potential user-provided values.

Reply:
Agreed with @fmigneault, defensive coding make the code more robust.